io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -766,7 +842,10 @@ reschedule !simstate@SimState{ threads, timers, curTime = time } =
                      , let Just vids = Set.toList <$> Map.lookup tid' wokeby ]
                   ++ [ ( time', tid, Just "thread delay timer"
                        , EventThreadDelayExpired)
-                     | tid <- wakeupThreadDelay ])
+                     | tid <- wakeupThreadDelay ]
+                  ++ [ ( time', tid, Just "timeout timer"
+                       , EventTimeoutFired tmid)
+                     | (tid, tmid, _, _) <- timeoutExpired' ])
```

Comment:
We also ought to record that we forked threads when we used `forkTimeoutInterruptThreads`.